kore/test/Test/Kore/Builtin/Bytes.hs

```diff
@@ -0,0 +1,522 @@
+module Test.Kore.Builtin.Bytes where
+
+import qualified Data.ByteString.Char8 as BS
+import Hedgehog hiding
+    ( Concrete
+    )
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Test.Tasty
+
+import Data.ByteString
+    ( ByteString
+    )
+import Data.Char
+    ( ord
+    )
+import Data.Text
+    ( Text
+    )
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as E
+import GHC.Stack
+    ( HasCallStack
+    )
+
+import qualified Kore.Builtin.Bytes as Bytes
+import Kore.Internal.Pattern
+import Kore.Internal.TermLike hiding
+    ( bytesSort
+    )
+
+import Test.Kore.Builtin.Builtin
+import Test.Kore.Builtin.Definition
+import qualified Test.Kore.Builtin.Int as Test.Int
+import qualified Test.Kore.Builtin.String as Test.String
+import Test.SMT
+
+genString :: Gen Text
+genString = Gen.text (Range.linear 0 256) Gen.alphaNum
+
+genString' :: Int -> Gen Text
+genString' i = Gen.text (Range.linear 0 i) Gen.alphaNum
+
+test_bytes2string :: TestTree
+test_bytes2string =
```

Comment:
I don't think that `test_bytes2string` and `test_string2bytes` are useful tests individually, especially because they are tested to be inverses of each other below.